### PR TITLE
[docs] Add a confirmation dialog demo

### PR DESCRIPTION
**Pull request by vladmoroz**
https://deploy-preview-1164--base-ui.netlify.app/react/components/dialog#close-confirmation

### CODE REVIEW

File: docs/src/app/(public)/(content)/react/components/dialog/page.mdx [hunk 1]
```diff
@@ -98,4 +98,12 @@ You can nest dialogs within one another normally.
 
 Use the `[data-has-nested-dialogs]` selector and the `var(--nested-dialogs)` CSS variable to customize the styling of the parent dialog. Backdrops of the child dialogs won't be rendered so that you can present the parent dialog in a clean way behind the one on top of it.
 
-<Demo path="./demos/nested" />
+<Demo compact path="./demos/nested" />
+
+### Close confirmation
+
+This example shows a nested confirmation dialog that opens before the parent dialog may close.
```
[the remark this answers is not in the document]
Comment (vladmoroz) [reply docs/src/app/(public)/(content)/react/components/dialog/page.mdx]: Makes sense, updated

**Comment by vladmoroz** [comment]
@colmtuite I updated to use an alert dialog, but we'll need to wait for the fix to land because the nested dialog styles are now broken

**Comment by vladmoroz** [comment]
Also adding the same demo to the alert dialog page

**Review by colmtuite — APPROVED** [review]

**Comment by colmtuite** [comment]
@vladmoroz Much better. Approved for whenever we get the alertdialog fix ✅

**Comment by mnajdova** [comment]
@vladmoroz https://github.com/mui/base-ui/pull/1167 has been merged

**Pull request merged**